include/tuvx/linear_algebra/linear_algebra.inl

```diff
@@ -1,99 +1,116 @@
 
-namespace tuvx {
+namespace tuvx
+{
 
-template <typename T>
-inline TridiagonalMatrix<T>::TridiagonalMatrix(std::size_t size) {
-  this->size_ = size;
-  this->main_diagonal_ = std::vector<T>(size);
-  this->upper_diagonal_ = std::vector<T>(size - 1);
-  this->lower_diagonal_ = std::vector<T>(size - 1);
-}
+  template<typename T>
+  inline TridiagonalMatrix<T>::TridiagonalMatrix(std::size_t size)
+  {
+    this->size_ = size;
+    this->main_diagonal_ = std::vector<T>(size);
+    this->upper_diagonal_ = std::vector<T>(size - 1);
+    this->lower_diagonal_ = std::vector<T>(size - 1);
+  }
 
-template <typename T>
-inline std::vector<T> Dot(const TridiagonalMatrix<T> &A,
-                          const std::vector<T> &x) {
-  std::size_t size = x.size();
-  std::vector<T> v(size);
-  v[0] = A.main_diagonal_[0] * x[0] + A.upper_diagonal_[0] * x[1];
+  template<typename T>
+  inline std::vector<T> Dot(const TridiagonalMatrix<T> &A, const std::vector<T> &x)
+  {
+    std::size_t size = x.size();
+    std::vector<T> v(size);
+    v[0] = A.main_diagonal_[0] * x[0] + A.upper_diagonal_[0] * x[1];
 
-  std::size_t i = 0;
-  for (i = 1; i < size - 1; i++) {
-    v[i] = A.main_diagonal_[i] * x[i] + A.upper_diagonal_[i] * x[i + 1] +
-           A.lower_diagonal_[i - 1] * x[i - 1];
+    std::size_t i = 0;
+    for (i = 1; i < size - 1; i++)
+    {
+      v[i] = A.main_diagonal_[i] * x[i] + A.upper_diagonal_[i] * x[i + 1] + A.lower_diagonal_[i - 1] * x[i - 1];
+    }
+    v[i] = A.main_diagonal_[i] * x[i] + A.lower_diagonal_[i - 1] * x[i - 1];
+    return v;
   }
-  v[i] = A.main_diagonal_[i] * x[i] + A.lower_diagonal_[i - 1] * x[i - 1];
-  return v;
-}
 
-template <typename T>
-inline std::vector<T> Solve(TridiagonalMatrix<T> A, std::vector<T> b) {
-  T temp;
-  std::size_t N = b.size();
-  std::vector<T> x(N);
-  // forward pass
-  for (std::size_t i = 1; i < N; i++) {
-    temp = A.lower_diagonal_[i - 1] / A.main_diagonal_[i - 1];
-    A.main_diagonal_[i] -= temp * A.upper_diagonal_[i - 1];
-    b[i] -= temp * b[i - 1];
-  }
-  // back substitution
-  x[N - 1] = b[N - 1] / A.main_diagonal_[N - 1];
-  for (std::size_t i = N - 2;; i--) {
-    x[i] = (b[i] - A.upper_diagonal_[i] * x[i + 1]) / A.main_diagonal_[i];
-    if (i == 0) {
-      break;
+  template<typename T>
+  inline std::vector<T> Solve(TridiagonalMatrix<T> A, std::vector<T> b)
```

Comment:
Just curious, could this be done in place? Could we reuse `b` and not make `x`?

Reply:
Yes, we can. I will make this change in my next commit.